From 26dca2c4030d5fd7511ec093f9e00a112c435afe Mon Sep 17 00:00:00 2001
From: Mazhar <mazharx.sameullah@intel.com>
Date: Wed, 9 Jan 2013 19:26:08 +0530
Subject: [PORT FROM MAIN] Loss of headphone audio when HDMI plug-in

BZ: 74544

In setOutputDevice function, if the device selected is AUDIO_DEVICE_NONE, then
updating the force value to false. In JB 4.2 release whenever the device is
disconnected, force is applied for the new device selection, even if the device
is none.

Change-Id: I9a0c7b7d9c15d86618200c46ea4ce68028e7f178
Orig-Change-Id: Ibfda806579014b26fd47081c313eb74f1cc52a2c
Signed-off-by: Mazhar <mazharx.sameullah@intel.com>
Reviewed-on: http://android.intel.com:8080/85826
Reviewed-by: Anand, Jerome <jerome.anand@intel.com>
Reviewed-by: Sikkandar D, Madar <madar.sikkandar.d@intel.com>
Tested-by: P C, SreekanthX <sreekanthx.p.c@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 audio/AudioPolicyManagerBase.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index 77692cf..f2d9ace 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -2442,6 +2442,8 @@ uint32_t AudioPolicyManagerBase::setOutputDevice(audio_io_handle_t output,
         // Device is null and does not reflect the routing. Save the necessity to force
         // re-routing upon next attempt to select a non-null device for this output
         outputDesc->mForceRouting = true;
+        // When the current device is null, routing is not forced
+        force = false;
     }
 
     muteWaitMs = checkDeviceMuteStrategies(outputDesc, prevDevice, delayMs);
-- 
1.7.4.1

